### DISCUSSION TITLE
How do we daemonize Next.js

### DISCUSSION THREAD

**Discussion by fvlasie** [vercel/next.js#21710] [Help] [answered]
How do we get Next.js to run as a systemctl controlled process so that it can survive restarts etc. ?

Thank you!

**Comment by fvlasie** [comment]
Similar to this for node.js : https://stackoverflow.com/questions/4018154/how-do-i-run-a-node-js-app-as-a-background-service/29042953#29042953

**Comment by jamesmosier** [comment, marked as answer]
You can use [PM2](https://pm2.keymetrics.io/) which is a great choice for Node applications such as Next.js.

Here are a few other discussions on it:

https://github.com/vercel/next.js/discussions/19728
https://github.com/vercel/next.js/discussions/10675

**Reply by fvlasie** [reply]
Thank you!
